### ISSUE TITLE
Read const amount of rows from a column

### ISSUE THREAD

**Issue by YuvalGrMic** [aloneguid/parquet-dotnet#353]
### Issue description

Hi,
I can't find a way where I can read up to a const (i.e K=100) rows in each column:
`                        var columnData = await groupReader.ReadColumnAsync();
`
I am using the Lib to sample K rows from each column and since there is no limitation on the number of rows that can be in each row Group, this can mean I will read N rows where: N>>K, and that can be a problem for my service.
Can I limit the number of rows I am reading from each column? (under the assumption that I have no control over the files I am reading).
Is there a plan to support this feature?
I really appreciate any help you can provide.
Yuval

**Comment by aloneguid** [comment]
No, this is not possible as parquet format doesn't store individual values but rather entire compressed chunks.

**Comment by YuvalGrMic** [comment]
These compressed chunks are of an entire RowGroup column? 
If no, are we able to return a single chunk at a time, i.e. at least part of the rows in the column?

Is there any Size limit for these chunks? 
Do you know if other libs are offering an option to sample K rows from a column?

Again thanks in advance!

**Comment by aloneguid** [comment]
There is physically no option to sample rows n **any** library, this is how parquet format works. The data is stored in columns, not rows.

For instance, in row-based formats, like CSV, in order to sample a column you would need to read all rows. In parquet, being column-based, the reverse applies.

You can, however, read a *column chunk*, but only if data was written in chunks in the file itself:

```csharp
using(Stream fs = System.IO.File.OpenRead("/mnt/storage/data.parquet")) {
    using(ParquetReader reader = await ParquetReader.CreateAsync(fs)) {
          // open the first row group (0) - reads the first chunk
          using(ParquetRowGroupReader rowGroupReader = reader.OpenRowGroupReader(0)) {

                  DataColumn columnData = await rowGroupReader.ReadColumnAsync(/* column */);

          }
    }
}
